### PR TITLE
📖 Define proposal guidelines and policies

### PR DESCRIPTION
**Pull request by vincepri**
This change tries to define some basic rules and guidelines on how we
usually handle the proposal process and how it all connects to
pre-defined releases.

/milestone v0.4.0

### CODE REVIEW

File: CONTRIBUTING.md
Comment (fabriziopandini) [inline CONTRIBUTING.md:82]: Might be add that a CAEP could be required also for changes without impacts on the API (e.g cluterctl, clusterctl templating)

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/hold
For others to review

File: CONTRIBUTING.md
Comment (neolit123) [inline CONTRIBUTING.md:64]: `twice / calendar year.` -> `twice in a calendar year.`

maybe, if MAJOR versions are not planned yet, i'd keep the above sentence about the MINOR cadence and include a note that cadence of MAJOR versions is TBD - e.g. after `1.` is released.

**Comment by neolit123** [comment]
reading https://github.com/kubernetes-sigs/cluster-api/blob/c997c1e5cc64b1f70a6922b9984b914bb49fed30/VERSIONING.md

should:
> an X (major) release indicates a set of backwards-compatible code. Changing X means there's a breaking change.

be `a set of backwards-incompatible` code?

**Comment by vincepri** [comment]
Yeah VERSIONING needs some updates separately, maybe we can find a better version available upstream

**Comment by nader-ziada** [comment]
lgtm

File: CONTRIBUTING.md
Comment (detiber) [reply CONTRIBUTING.md:64]: +1, let's avoid talking about major releases until we have a need to.

File: CONTRIBUTING.md
Comment (detiber) [reply CONTRIBUTING.md:82]: Along the lines of @fabriziopandini's comment:
```suggestion
The Cluster API Enhacement Proposal is the process this project uses to adopt new features, changes to the APIs, changes to contracts between components, or changes to CLI interfaces.
```

File: CONTRIBUTING.md
Comment (detiber) [inline CONTRIBUTING.md:86]: Can this be reworded to something along the lines of: Proposals MUST have an associated issue? That way it doesn't sound like we are gate keeping folks from trying to write a proposal if they feel that it would more clearly state their case than doing so in an issue directly.

That way it is clear that the issue is required for being included in a release rather than being a prerequisite step for someone starting to put together a proposal if they want to.

File: CONTRIBUTING.md
Comment (detiber) [inline CONTRIBUTING.md:91]: I'd like to see us do a better job of broadcasting proposals than we do currently. If someone isn't watching the repository and attending all the meetings it is difficult to stay on top of proposed changes. I'm not necessarily saying it should be the responsibility of the person creating the proposal, but as a project we should do a better job of updating the mailing list and discuss forum know about proposals with a decent lead time for folks to review and provide feedback.

File: CONTRIBUTING.md
Comment (vincepri) [reply CONTRIBUTING.md:86]: Good feedback! Thanks, reworded

File: CONTRIBUTING.md
Comment (vincepri) [reply CONTRIBUTING.md:91]: How would you like for this to change? We can make it maintainer's responsibilities?

Comment (detiber) [reply CONTRIBUTING.md:91]: I think making it the maintainer's responsibility makes sense to me, the main goal to allow folks to more easily participate and keep track of proposed changes without having to fully drink from the firehose.

Comment (yastij) [reply CONTRIBUTING.md:91]: today for the kubernetes community, the goto repo for proposals is `k/enhancements` at some point we might want have `CAEPs` live next to `KEPs` under the sig cluster-lifecycle folder. I think this would also broaden the audience of the project. thoughts ?

Comment (neolit123) [reply CONTRIBUTING.md:91]: there is a folder for CAPI KEPs here:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/clusterapi

we should PR a README.md to that folder to point to the k-sigs/cluster-api folder for CAEPs.

if CAEPs start living in k/e, at minimum we would have to auto-apply the `tracked/out-of-tree` label to the CAPI folder to try to reduce the noise for the release team, who normally track k8s cadence proposals.

Comment (detiber) [reply CONTRIBUTING.md:91]: k/enhancements is it's own firehose, not saying we couldn't standardize on using it, I'm just not sure if it would achieve better community visibility than we have today.

Comment (vincepri) [reply CONTRIBUTING.md:91]: `k/e` Cluster API proposal seems very out of date 😬, we probably need to add a minimum of text that says we've our own process. I'm in favor of keeping the process in house though, rather than using the broader process

Comment (randomvariable) [reply CONTRIBUTING.md:91]: I don't see a benefit of using k/e, as maintainers of the k/e process aren't (understandably) going to be invested in process improvements for us.

Comment (neolit123) [reply CONTRIBUTING.md:91]: > Cluster API proposal seems very out of date

`+`

> there is a folder for CAPI KEPs here...(snip)...we should PR a README.md to that folder (in k/e)

i will log a ticket for these two.

Comment (neolit123) [reply CONTRIBUTING.md:91]: https://github.com/kubernetes-sigs/cluster-api/issues/3791

**Review by devigned — APPROVED** [review]
/lgtm

File: CONTRIBUTING.md
Comment (yastij) [reply CONTRIBUTING.md:91]: sounds good to me, I was mainly asking for discoverability

**Pull request merged**
